### DISCUSSION TITLE
Question about link rendering

### DISCUSSION THREAD

**Discussion by TiLied** [dotnet/docfx#8961] [Q&A] [answered]
Hello. I have a question, why xml code like this
`<seealso href="https://developer.mozilla.org/en-US/docs/Web/API/AbortController"><em>See also on MDN</em></seealso>`
renders like so
`<a href="https://developer.mozilla.org/en-US/docs/Web/API/AbortController" target="_blank" rel="noopener noreferrer nofollow" class="external"><em>See also on MDN</em>https://developer.mozilla.org/en-US/docs/Web/API/AbortController</a>`
![screan](https://github.com/dotnet/docfx/assets/26207264/2834eba9-59f7-4de2-b6b8-5309c3615c32)
Can it be changed? I couldn't find any documentation on this.

It renders correctly in Visual Studio.
![screen2](https://github.com/dotnet/docfx/assets/26207264/27d94769-f196-4030-aab9-29f068e55758)

**Comment by KalleOlaviNiemitalo** [comment, marked as answer]
Looks like a bug in <https://github.com/dotnet/docfx/pull/1157>: <https://github.com/dotnet/docfx/blob/d90b31868912523be1070bdf4e7bc0581119535b/src/Docfx.Dotnet/Resources/XmlCommentTransform.xsl#L28-L44>

Because your `seealso` element has no text directly in it (although there is text in the `em` child element), this template assumes there is nothing visible and outputs the href as content.  The template for `see` has the same problem.

The templates could probably be fixed by replacing `test="not(text())"` with `test="not(node())"`.  Alternatively, `test="not(.//text())"` would find text within child elements too, but it would still treat `<seealso href="http://example.org/"><img src="http://example.org/"/></seealso>` as empty.

Until that is fixed, you may be able to work around the bug by adding whitespace before `<em>` or after `</em>`.

**Reply by TiLied** [reply]
Thank you) Workaround worked for me.

**Reply by KalleOlaviNiemitalo** [reply]
Filed the bug as <https://github.com/dotnet/docfx/issues/8965>.
